### PR TITLE
[W08H03] Add a test with a tree where a node returns `[null, a, null,  b]` for `getChildren()`

### PR DESCRIPTION
**Pull request by nilsreichardt**
image

With this tree the node 13 returns `[null, 14, null, 15]` when calling `getChildren()`. My first implementation passed all other tests but failed this one. Therefore, I'm adding it to our tests.

### CODE REVIEW

File: w08h03/test/pgdp/datastructures/UnitTest.java [hunk 1]
```diff
@@ -48,6 +48,15 @@ public void artemisTest() {
 		testArray(new Integer[] { 8, 4, 12, 1, 5, 9, 13, 3, 7, 11, 15, 2, 6, 10, 14 });
 	}
 
+	@Test
+	@DisplayName("should iterate over graph where a node can have 'null, a, null b' as children")
+	public void testChildrenContainingNull() {
+		// The value 14 and 15 are the children of the node 13. However, returns
+		// `getChildren` of node 13 in the following order: [null, 14, null,
+		// 15]. Therefore, your implementation should handle this case. 
+		testArray(new Integer[] { 18, 8, 4, 12, 1, 5, 9, 13, 3, 7, 11, 15, 2, 6, 10, 14 });
```
Comment (Anatoly03) [inline w08h03/test/pgdp/datastructures/UnitTest.java:57]: This testArray method looks really cool. The guy who made it must be excellent 🐧

**Review by Anatoly03 — APPROVED** [review]
🐧

**Review by MaximilianAnzinger — COMMENTED** [review]
[🐧](https://emojipedia.org/penguin/)
